File: app/src/main/java/com/example/tracker/weight/FirestoreWeightRepository.java
package com.example.tracker.weight;

import com.google.firebase.firestore.QueryDocumentSnapshot;

import java.util.HashMap;
import java.util.Map;

public class FirestoreWeightRepository {

    public static WeightDto from(QueryDocumentSnapshot document) {
        long timeInMillis = (long) document.getData().get("timeInMillis");
        double weightInKgs = (double) document.getData().get("weight in kilograms");
        return new WeightDto(timeInMillis, weightInKgs);
    }

    public static Map<String, Object> toMap(WeightDto dto) {
        Map<String, Object> map = new HashMap<>();
        map.put("timeInMillis", dto.getTimeInMillis());
        map.put("weight in kilograms", dto.getWeightInKgs());
        return map;
    }

    public static String getId(WeightDto dto) {
        return dto.getTimeInMillis() + "_" + dto.getWeightInKgs();
    }
}

File: app/src/main/java/com/example/tracker/weight/FirestoreWeightRepository.kt
package com.example.tracker.weight

import com.google.firebase.firestore.QueryDocumentSnapshot

object FirestoreWeightRepository {

    @JvmStatic
    fun from(document: QueryDocumentSnapshot): WeightDto {
        val timeInMillis = document.data["timeInMillis"] as Long
        val weightInKgs = document.data["weight in kilograms"] as Double
        return WeightDto(timeInMillis, weightInKgs)
    }

    @JvmStatic
    fun toMap(dto: WeightDto): Map<String, Any> {
        val map: MutableMap<String, Any> = HashMap()
        map["timeInMillis"] = dto.timeInMillis
        map["weight in kilograms"] = dto.weightInKgs
        return map
    }

    @JvmStatic
    fun getId(dto: WeightDto): String {
        return dto.timeInMillis.toString() + "_" + dto.weightInKgs
    }
}